File: Coursera/Algorithms_Part_2/Ex19DirectedEdge.java
/* *****************************************************************************
 *  Name:
 *  Date:
 *  Description:
 **************************************************************************** */

public class Ex19DirectedEdge implements Comparable<Ex19DirectedEdge> {

    private final int v, w;
    private final double weight;

    public Ex19DirectedEdge(int v, int w, double weight) {
        this.v = v;
        this.w = w;
        this.weight = weight;
    }

    public int from() {
        return v;
    }

    public int to() {
        return w;
    }

    public double weight() {
        return weight;
    }

    public String toString() {
        return v + " -- " + weight + " --> " + w;
    }

    public int compareTo(Ex19DirectedEdge that) {
        return Double.compare(this.weight(), that.weight());
    }

    public static void main(String[] args) {

    }
}

File: PythonLearnig/Algorithms_Part_2/Ex19DirectedEdge.py
class DirectedEdge:

    def __init__(self, v, w, weight):
        self.v = v
        self.w = w
        self.weight = weight

    def from_v(self):
        return self.v

    def to_v(self):
        return self.w

    def get_weight(self):
        return self.weight

    def compare_to(self, that):
        return self.get_weight() - that.get_weight()

    def __str__(self):
        return f'{self.from_v()} -- {self.weight} --> {self.to_v()}'


def test_directed_graph():
    print(DirectedEdge(3, 7, 10.7))


if __name__ == '__main__':
    test_directed_graph()
